crypto/asn1/x_long.c

```diff
@@ -76,7 +76,7 @@ static int long_i2c(ASN1_VALUE **pval, unsigned char *cont, int *putype,
      * set.
      */
     if (ltmp < 0)
-        utmp = -(unsigned long)ltmp - 1;
+        utmp = 0 - (unsigned long)ltmp - 1;
```

Comment:
I think this is actually fine, the 0 will get both promoted and converted to the unsigned long before it does the 0 - (unsigned long)ltmp, and you'll get a result of an unsigned long. And then you'll do that -1 after the one gets changed to the unsigned long.